apps/src/blockly/customBlocks/googleBlockly/mutators/behaviorDefMutator.js

```diff
@@ -0,0 +1,206 @@
+// import {CdoParameterModel} from './parameterModel';
+import {ObservableParameterModel} from '@blockly/block-shareable-procedures';
```

Comment:
this file is very similar to `procedureDefMutator`. Can we consolidate some of the logic?

Reply:
Since the plan is to remove `procedureDefMutator` once we've upgraded Blockly to v10 (and my understanding is that that upgrade isn't a huge lift), is it worth punting on a refactor?

I assume we'll need the `behaviorDefMutator` even after the v10 upgrade, but do you have any more insight here? The `domToMutation` function is very different, and there is an extra line each in `saveExtraState` and `loadExtraState`, but otherwise, they're the same.

Reply:
ah good point. I'm not sure about whether or not we'll need this one going forward, we might because it has a special `domToMutation` function. There may not be an equivalent blockly version of this since this is more specialized.